src/components/MapStyleInputWrapper.svelte

```diff
@@ -0,0 +1,180 @@
+      }
+    }
+
+    selectedValue = nextValue;
+  };
+
+  // Creates dropdown values and display options for the style dropdown linked by ids
+  const setInitialDropdownOptions = stylePresets => {
+    // Create values and displays for style presets
+    if (stylePresets.length) {
+      const stylePresetValues = stylePresets.map(item => ({
+        ...item,
+        dropdownType: 'preset',
+        selected: url === item?.url,
+        dropdownId: hat(),
+      }));
+
+      dropdownValues = dropdownValues.concat(stylePresetValues);
+
+      dropdownDisplayOptions['Presets'] = stylePresetValues.map(item => ({
+        text: item.name,
+        dropdownId: item.dropdownId,
+      }));
+    }
+
+    // Create values and displays for branch options
+    if (branchPatterns) {
+      for (const pattern of branchPatterns) {
+        if (pattern?.styles?.length) {
+          const branchValues = pattern?.styles.map(s => {
+            return {
+              name: `${s.charAt(0).toUpperCase() + s.slice(1)} on...`,
+              id: s,
+              type: pattern.type,
+              dropdownType: 'branch',
+              selected: !!(
+                branch && createBranchUrl(pattern.pattern, branch, s) === url
+              ),
+              pattern: pattern.pattern,
+              dropdownId: hat(),
+            };
+          });
+
+          dropdownValues = dropdownValues.concat(branchValues);
+
+          dropdownDisplayOptions[
+            `Styles on a branch${pattern.name ? `: ${pattern.name}` : ''}`
+          ] = branchValues.map(item => ({
+            text: item.name,
+            dropdownId: item.dropdownId,
+          }));
+        }
+      }
+    }
+
+    // If no option has matched the URL, then it is custom
+    const hasSelectedOption = dropdownValues.some(item => !!item.selected);
+
+    // Create a custom value and display option
+    const customValues = [
+      {
+        name: 'Fetch URL at...',
+        dropdownType: 'custom',
+        selected: !hasSelectedOption,
+        type: 'mapbox-gl',
+        dropdownId: hat(),
+      },
+    ];
+
+    dropdownValues = dropdownValues.concat(customValues);
+
+    dropdownDisplayOptions['Custom'] = customValues.map(item => ({
+      text: item.name,
+      dropdownId: item.dropdownId,
+    }));
+  };
+
+  const setInitialSelectedOption = stylePresets => {
+    setInitialDropdownOptions(stylePresets);
+    setSelectedValue();
+  };
+
+  // We can't do this onMount because the stylePresets store will
+  // update more than once if there's style preset URLs
+  stylePresetsStore.subscribe(value => {
+    setInitialSelectedOption(value);
+  });
+
+  const onSelectOption = e => {
+    const { dropdownId } = e.detail;
+    // Set selected property on value
+    dropdownValues = dropdownValues.map(v => {
+      if (v.dropdownId === dropdownId) {
+        return { ...v, selected: true };
+      } else {
+        return { ...v, selected: false };
+      }
+    });
```

Comment:
Any reason not to go with a more compact version like this?

```suggestion
    dropdownValues = dropdownValues.map(v => {
      return { ...v, selected: v.dropdownId === dropdownId };
    });
```

Reply:
Oh yea that's better. Just wrote this long form as I was thinking through things haha.